akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -63,6 +74,22 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
   def wrapTrailingHeaders(headers: ParsedHeadersFrame): Option[ByteString] = None
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[http2] object Keepalive {
+  case object Tick
+  val Ping = PingFrame(false, ByteString("abcdefgh"))
+  final class PingState(val maxKeepaliveTimeoutNanos: Long, val pingEveryNTickWithoutData: Long) {
```

Comment:
Good to move this into an extra class. Could we move all of the ping functionality in there? Especially mutating the vars?

Reply:
Sure, I'll give it a try.